hw/dv/sv/cip_lib/cip_lib.core

```diff
@@ -9,6 +9,7 @@ filesets:
   files_dv:
     depend:
       - lowrisc:dv:dv_lib
+      - lowrisc:dv:dv_base_reg
```

Comment:
I could be wrong, but I thought if `lowrisc:dv:dv_lib` has already import this core file, we do not need to import it again here. (similar reason as we de not import `csr_util_pkg` again)?

Reply:
Good catch. Fixed. Thanks

Reply:
Actually, it is better to have fusesoc core files add all required dependencies correctly than infer / assume a dependency hierarchy.

Reply:
but usually we don't specify the the sub core files if it's specified in the top hierachy.
For instance, sim.core only specifies the test.core. Test.core specifies env.core. Env.core specifies many dependent common core files.

Reply:
Right, let me clarify. If A->B->C then A adds dependency on B and B adds dependency on C . If A does not import / use stuff directly from C, then we are good to go. 

However, if A -> (B, C) and B -> C, then A needs to add dependency on both, B and C, rather than assume that the B -> C will take care of the ordering. 

In this case, the `cip_lib` uses stuff from `dv_lib` as well as `dv_base_reg`. So rather than assume `dv_lib` takes care of `dv_base_reg` dependency, `cip_lib` should also add dependency on `dv_base_reg`. This is a recent recommendation from Phillip to ensure that the fusesoc dependency solver works correctly.